### PR TITLE
NodeAttack & EdgeAttack

### PR DESCRIPTION
**Pull request by annacanal**
#289 Edgeattack test not always working yet; Nodeattack test working

### CODE REVIEW

File: braph2/graph/graphs/GraphBD.m [hunk 1]
```diff
@@ -81,11 +81,11 @@
         function graph_type = getGraphType()
             graph_type = Graph.GRAPH;
         end
-        function graph_type = getConnectionType()
-            graph_type = Graph.BINARY;
+        function connection_type = getConnectionType()
```
Comment (egolol) [inline braph2/graph/graphs/GraphBD.m:84]: will it change to connectivity?

Comment (annacanal) [reply braph2/graph/graphs/GraphBD.m:84]: If we want to change it to connectivity I will do it in another issue for that.

**Comment by annacanal** [comment]
I think this is good now.
For edgeattack in undirected graphs, I kept the addition of removing the corresponding edge but checking that the matrix is square. I think is much nicer to do that, that the user having to introduce double number of edges to remove for undirected graphs.

**Comment by annacanal** [comment]
> Nodeattack seems nice.
> 
> I think I misunderstood something regarding the edge attacks so, could I ask you for a short explanation.
> 
> These are the relevant comments from the edgeattack section:
> % GA = EDGEATTACK(G, NODES1, NODES2, LAYERNUMBERS_I, LAYERNUMBERS_J)
> % creates the graph GA resulting by removing the edges going
> % from NODES1 to NODES2 from G. For non single layer graphs, it
> % removes the edges from NODES1 to NODES2 in the layers
> % specified by LAYERNUMBERS.
> From here I understand that edges are to removed in multiple layers (that are specified by an input) between NODES1 and NODES2. In that case, why do we need two inputs to specify layers? If a layer can be got from the graph as A{i, i}, I am not sure why we need two inputs. Or do you in this way account for edges that run between layers?

Exactly, when we have I and J is in order to account for in between layers

**Comment by annacanal** [comment]
> > Nodeattack seems nice.
> > I think I misunderstood something regarding the edge attacks so, could I ask you for a short explanation.
> > These are the relevant comments from the edgeattack section:
> > % GA = EDGEATTACK(G, NODES1, NODES2, LAYERNUMBERS_I, LAYERNUMBERS_J)
> > % creates the graph GA resulting by removing the edges going
> > % from NODES1 to NODES2 from G. For non single layer graphs, it
> > % removes the edges from NODES1 to NODES2 in the layers
> > % specified by LAYERNUMBERS.
> > From here I understand that edges are to removed in multiple layers (that are specified by an input) between NODES1 and NODES2. In that case, why do we need two inputs to specify layers? If a layer can be got from the graph as A{i, i}, I am not sure why we need two inputs. Or do you in this way account for edges that run between layers?
> 
> Exactly, when we have I and J is in order to account for in between layers

Oh I see I didnt finished properly the comment in that case, I left the previous one. I will do a commit changing this

**Pull request merged**